src/components/SubmitIdea.jsx

```diff
@@ -0,0 +1,222 @@
+import React, { useState, useEffect } from 'react';
+import { Container, Form, Button, Row, Col } from 'react-bootstrap';
+import { Typeahead } from 'react-bootstrap-typeahead';
+import 'react-bootstrap-typeahead/css/Typeahead.css';
+import axios from 'axios';
+
+const SubmitIdea = () => {
+  const [retrievedTechStacks, setRetrievedTechStacks] = useState([]);
+
+  const [name, setName] = useState('');
+  const [description, setDescription] = useState('');
+  const [why, setWhy] = useState('');
+  const [techStack, setTechStack] = useState([]);
+  const [teamNumber, setTeamNumber] = useState('');
+  const [imageURL, setImageURL] = useState('');
+  const [whenStart, setWhenStart] = useState('');
+  const [whenEnd, setWhenEnd] = useState('');
+  const [isSubmitted, setIsSubmitted] = useState(false);
+
+  // get request to backend to fetch tech stack data
+  useEffect(() => {
```

Comment:
Note to self: a functional component's `componentDidMount`  equivalent (but even more flexible as to configuration)